### ISSUE TITLE
Please get "Fix Model" working in the linux version

### ISSUE THREAD

**Issue by crackinx** [SoftFever/OrcaSlicer#5013]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

The only problem it is related to is that Fix Model doesn't work under Linux.

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

As the title says, I don't run windows and am not versed enough in cad to find non manifold edges myself.

**Comment by mlewis-everley** [comment]
This would be super handy, especially as Orca has simple boolean operations available now.

**Comment by buzzhuzz** [comment]
I've actually looked at this few weeks back and found that on windows model is fixed by Windows SDK component.

My approach would be is to bring integration with [MeshLab](https://www.meshlab.net/) using [PyMeshLab](https://pymeshlab.readthedocs.io/en/latest/) 

However, I've put this idea on the shelf as it is not causing any real issues for me currently

**Comment by algocrypto** [comment]
Also can this be implemented on Mac OSX as well?

**Comment by neclimdul** [comment]
Unnecessary noise because bot.

**Comment by aidan-gibson** [comment]
Should be merged with #4923 or vice versa.

(Fuck windows aaaaaaa)

**Comment by D0m-Sch** [comment]
I second that. Would be really nice to have.

**Comment by catgirlcataclysm** [comment]
I agree, ive had immense amounts of trouble getting anything to work on the windows partition on my laptop, so it would be great if there was a way to do this natively from my desktop (linux)

**Comment by xznhj8129** [comment]
I am not booting windows just for this.

**Comment by JamesOsborn-SE** [comment]
> I am not booting windows just for this.

If you need something today it's a workaround, but I've had great success using Blender and 3D Print Toolbox Add-on/extension. I still hope they will add this feature, so I don't have to rely on other tools.

**Comment by blipside** [comment]
I haven't used Window for over 15 years now.  Is this something that the team is going to work on?  It is annoying that the Calibration features need to be fixed.  I am using v2.3.0, Build 70931e5.  Thank you.
Temp Calibration
Error: 14 non-manifold edges
Tips: "Fix Model" feature is currently only on Windows.  Please repair the model on Orca Slicer(windows) or CAD software.

**Comment by LAP87** [comment]
Please add this feature, i left windows a few months ago and every day there are more migrants over to the linux side of life.

**Comment by blipside** [comment]
When will this feature be added?  I haven't used Windows for over 15 years.

**Comment by sperdev54** [comment]
Please make this feature cross-platform. It’s very useful, and since I’m already on Linux, I can’t use Microsoft 3D Builder, which was also great for quick fixes.

**Comment by blipside** [comment]
At least make it accept an alternative to Microsoft 3D Builder, like OpenSCAD.  OpenSCAD is available for Linux/UNIX, MS Windows, and MacOS.

**Comment by coryrc** [comment]
> When will this feature be added? I haven't used Windows for over 15 years.

When you figure out how to do it and implement the code.

**Comment by sperdev54** [comment]
> When will this feature be added?  I haven't used Windows for over 15 years.

On Windows, there is a program for quick fixes on models called Microsoft 3D Builder. This program doesn’t offer anything professional, but for quick fixes it’s better than the automatic tools in Meshlab and Blender. From what I know, Orca Slicer uses this, so porting it to Linux is very difficult. At best, they could use the Meshlab API and make it work with that.

**Comment by jokeyrhyme** [comment]
huh, PrusaSlicer requires Windows for this as well: https://github.com/prusa3d/PrusaSlicer/issues/9066

**Comment by mlewis-everley** [comment]
> > When will this feature be added?  I haven't used Windows for over 15 years.
> 
> On Windows, there is a program for quick fixes on models called Microsoft 3D Builder. This program doesn’t offer anything professional, but for quick fixes it’s better than the automatic tools in Meshlab and Blender. From what I know, Orca Slicer uses this, so porting it to Linux is very difficult. At best, they could use the Meshlab API and make it work with that.

Unfortunately Microsoft 3D builder is considered discontinued now (although I don't think Microsoft has officially declared this), a lot of users cannot download it, even if they have previously.

The Fix 3D Model feature was (I think) implemented by Prusa, Not Bambu or Orca. Looks like it is called here: https://github.com/prusa3d/PrusaSlicer/blob/f1776c0a6347bb84986d10eac8db1021f5bd8548/src/slic3r/Utils/FixModelByWin10.cpp#L216

Looks like PrusaSlicer and derivatives (Orca appears to implement this here: https://github.com/SoftFever/OrcaSlicer/blob/7736b31015bdf1a2b3efd2fbb2634f568fa5d2f0/src/slic3r/Utils/FixModelByWin10.cpp#L213C20-L213C21) use Microsoft's `Printing3D` API: https://learn.microsoft.com/en-us/uwp/api/windows.graphics.printing3d

As mentioned previously, I guess the developers would have to rip all this code out and swap over to implementing an open source cross platform integration with a tool like MeshLab/PyMeshLab (or some other solution), which would be a fair chunk of work.

If anyone is really desperate for this feature cross platform, then OrcaSlicer is an open source project, you could fork the project, spend some time getting up to speed with C++ (it is quite a complex language, but there are a lot of guides out there) and feel free to submit a pull request. I am sure the orca dev's would be really happy for any help :-)

**Comment by MagnusVesper** [comment]
Maybe a setting could be added for the "fix model" operation. One of the options can be "custom" allowing the user to enter an executable to run. This would give MacOS and Linux users a way to make it work without having to fork the project.

This would still require significant work, because it will need to export, then re-import the mesh. But once it is done, it would require little maintenance.

**Comment by ianalexis** [comment]
For you all, check this PR: https://github.com/OrcaSlicer/OrcaSlicer/pull/12155

You can test it following this guide: https://www.orcaslicer.com/wiki/How-to-download-PR-artifacts
